packages/tc-api-db-manager/db-connection.js

```diff
@@ -28,37 +28,49 @@ const getTimeoutRacePromise = timeout =>
 		);
 	});
 
+const runQueryWithMetrics = async ({
+	session,
+	runner,
+	isTransaction = false,
+}) => {
+	const start = Date.now();
+	metrics.count('neo4j.query.count');
+	let isSuccessful = false;
+	try {
+		const result = TIMEOUT
+			? await Promise.race([runner(), getTimeoutRacePromise(TIMEOUT)])
+			: runner();
+
+		isSuccessful = true;
+		return result;
+	} finally {
+		if (!isTransaction) {
+			session.close();
+		}
```

Comment:
In the transaction, the session must not close here. the session must be closed after all queries have been run because neo4j-driver will do commit/rollback with using this session